client/cloudevents/protocol/fake/client.go

```diff
@@ -0,0 +1,48 @@
+package fake
+
+import (
+	"context"
+	"fmt"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"github.com/cloudevents/sdk-go/v2/event"
+	"github.com/cloudevents/sdk-go/v2/protocol"
+)
+
+type CloudEventClient struct {
+	sentEvents     []event.Event
+	receivedEvents []event.Event
+}
+
+func NewCloudEventClient(receivedEvents ...event.Event) *CloudEventClient {
+	return &CloudEventClient{
+		sentEvents:     []event.Event{},
+		receivedEvents: receivedEvents,
+	}
+}
+
+func (c *CloudEventClient) Send(ctx context.Context, event event.Event) protocol.Result {
+	c.sentEvents = append(c.sentEvents, event)
+	return nil
+}
+
+func (c *CloudEventClient) Request(ctx context.Context, event event.Event) (*event.Event, protocol.Result) {
+	return nil, nil
+}
+
+func (c *CloudEventClient) StartReceiver(ctx context.Context, fn interface{}) error {
+	receiver, ok := fn.(func(ctx context.Context, evt cloudevents.Event))
+	if !ok {
+		return fmt.Errorf("unsupported receiver function %T", fn)
+	}
+
+	for _, evt := range c.receivedEvents {
+		receiver(context.TODO(), evt)
```

Comment:
Is this intent to use new context, instead of `ctx`?

Reply:
fixed, it should not use a new context